### PR TITLE
[Kernel] `compressed-tensors` marlin 24 support

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Adds an additional `CompressedTensorsScheme` to supoprt `w4a16` models with `24` sparsity, `CompressedTensors24`
- Add support for `w4a16` models with `24` sparsity quantized through [sparseml](https://github.com/neuralmagic/sparseml) and saved through [compressed-tensors](https://github.com/neuralmagic/compressed-tensors)
- Leverages the kernels (`gptq_marlin_24_gemm`) used for `gptq_marlin-24`

From Neural Magic, Co-authored by @alexm-neuralmagic

### CODE REVIEW

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py [hunk 1]
```diff
@@ -7,17 +7,20 @@
 from vllm.model_executor.layers.quantization.base_config import (  # noqa: E501
     QuantizationConfig)
 from vllm.model_executor.layers.quantization.compressed_tensors.schemes import (
-    CompressedTensorsScheme, CompressedTensorsW4A16,
+    CompressedTensors24, CompressedTensorsScheme, CompressedTensorsW4A16,
```
Comment (robertgshaw2-neuralmagic) [inline vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:10]: Can we call this W4A16?

File: vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a16_24.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    def __init__(self,
+                 strategy: str,
+                 num_bits: int,
+                 group_size: Optional[int] = None):
+        self.strategy = strategy
+        self.group_size = group_size
+        self.num_bits = num_bits
+        self.tile_size = 16
+
+        if self.strategy == "group" and self.group_size is None:
```
Comment (robertgshaw2-neuralmagic) [inline vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a16_24.py:27]: do this support both 4 bits and 8 bits?

File: vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py (hunk 1 above)
Comment (robertgshaw2-neuralmagic) [reply vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py:10]: We will have other 24 methods in the future

**Review by robertgshaw2-redhat — APPROVED** [review]

**Comment by comaniac** [comment]
Please rebase or merge the latest main branch to update AMD CI config and numpy version to pass the CI.

File: vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a16_24.py (hunk 2 above)
Comment (dsikka) [reply vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a16_24.py:27]: Just 4bits atm

**Review by mgoin — APPROVED** [review]

**Pull request merged**